xdsl/dialects/math.py

```diff
@@ -67,22 +193,13 @@ class AbsIOp(IRDLOperation):
     """
 
     name = "math.absi"
-    operand = operand_def(IntegerType)
-    result = result_def(IntegerType)
 
     traits = traits_def(Pure(), SameOperandsAndResultType())
 
-    def __init__(self, operand: Operation | SSAValue):
-        operand = SSAValue.get(operand)
-        super().__init__(operands=[operand], result_types=[operand.type])
-
 
 @irdl_op_definition
-class Atan2Op(IRDLOperation):
+class Atan2Op(FloatingPointLikeBinaryMathOperationWithFastMath):
     """
-    Syntax:
-    operation ::= ssa-id `=` `math.atan2` ssa-use `,` ssa-use `:` type
-
```

Comment:
Not fond of adding the syntax in the doc; this was done only partially for some of the `math` ops, and we don't do it anywhere else in the codebase.